changelog/unreleased/bugfixes/7245.md

```diff
@@ -0,0 +1 @@
+- Added experimental and temporary (i.e. it will be removed in one of the next releases) `OnlineRouteAlternativesSwitch` which requests online route when the current route is offline and automatically switches if such a route is found. It's designed for the case when platform's reachability API doesn't work reliably.
```

Comment:
The file name is wrong.

Reply:
Oops, fixing it in a followup PR: https://github.com/mapbox/mapbox-navigation-android/pull/7266